src/herder/TransactionQueue.cpp

```diff
@@ -380,13 +388,7 @@ TransactionQueue::canAdd(
             {
                 auto txResult = tx->createSuccessResult();
                 if (!tx->checkSorobanResourceAndSetError(
-                        mApp.getAppConnector(),
-                        mApp.getLedgerManager()
-                            .getSorobanNetworkConfigReadOnly(),
-                        mApp.getLedgerManager()
-                            .getLastClosedLedgerHeader()
-                            .header.ledgerVersion,
-                        txResult))
+                        *mValidationSnapshot, ledgerVersion, txResult))
```

Comment:
isn't ledgerVersion stored inside mValidationSnapshot already?

Reply:
Yep. I removed passing around `ledgerVersion` in 16979788576cf86a4314dfb96be3751ef4233d9b